lib/apparmortest.pm

```diff
@@ -667,6 +670,77 @@ sub yast2_apparmor_cleanup {
     upload_logs("$audit_log");
 }
 
+# Create a test profile with name contains '('
+sub create_a_test_profile_name_is_special {
+    my ($testfile, $str) = @_;
+
+    my $testfile2 = "$testfile" . "$str";
+    assert_script_run("cp $testfile $testfile2");
+    assert_script_run("aa-autodep $testfile2");
+    assert_script_run("ll /etc/apparmor.d/ | grep $str");
+}
+
+# Refer to "https://bugs.launchpad.net/apparmor/+bug/1848227"
+# to create a test profile with content "local include above the '}'",
+# then run "aa-complain, aa-disable, aa-enforce, aa-logprof"
+# to verify all the commands should be succeeded
+sub test_profile_content_is_special {
+    my ($self, $cmd, $msg) = @_;
+    my $test          = "test_profile";
+    my $test_profile  = "/etc/apparmor.d/usr.sbin." . "$test";
+    my $local_profile = "/etc/apparmor.d/local/usr.sbin.cupsd";
+
+    # Create an empty local profile under "/etc/apparmor.d/local/"
+    assert_script_run("rm -rf $local_profile");
+    assert_script_run("touch $local_profile");
+
+    # Create a test profile under "/etc/apparmor.d/"
+    assert_script_run("echo '/usr/sbin/cupsd {' > $test_profile");
+    assert_script_run("echo '}' >> $test_profile");
+    assert_script_run("echo '#include <local/usr.sbin.cupsd>' >> $test_profile");
+
+    # Run aa-* commands and check the output
+    if ("$cmd" eq "aa-logprof") {
+        validate_script_output("$cmd", sub { m/$msg/ });
+    } else {
+        validate_script_output("$cmd $test_profile", sub { m/$msg/ });
+    }
+    if ("$cmd" eq "aa-disable") {
+        # The profile will not be listed out if disabled
+        my $ret = script_run("aa-status | grep $test");
```

Comment:
Have you tried `assert_script_run '... | grep -v ...` ?

Reply:
> Have you tried `assert_script_run '... | grep -v ...` ?

If you mean `assert_script_run 'aa-status | grep -v $other-test-profiles-etc` it  is not I expect as the `$test` should not appear in the output of `aa-status` and `$other-test-profiles-etc` is changeful.